andela/noldor-ah-backend
# #161290921 Highlight and Comment on an Article

#### What does this PR do?
This PR adds highlighting functionality to the Authors Haven API. Highlighting can only be performed by logged in users.

When highlighting, you can decide to add a comment along with it.

#### Description of Task to be completed?
- write tests for the new feature
- add highlighting functionality to API
- add commenting on highlight functionality
- return highlights when returning an article

#### How should this be manually tested?
1. Create an article #13 
2. Copy the article id and make a post request following the format specified in the screenshots section
3. You should get a success message if the instructions are followed successfully
4. Get the article you just highlighted from #13
5. You should see your highlights returned along with the article

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#161290921

#### Screenshots (if appropriate)
<img width="1132" alt="screen shot 2018-11-19 at 10 52 46 pm" src="https://user-images.githubusercontent.com/36507972/48737306-f6ad7580-ec4d-11e8-822d-8388a470cb2d.png">
<img width="1126" alt="screen shot 2018-11-19 at 10 50 17 pm" src="https://user-images.githubusercontent.com/36507972/48737398-2d838b80-ec4e-11e8-882c-7c6f48f9f8d3.png">
<img width="1129" alt="screen shot 2018-11-19 at 10 54 54 pm" src="https://user-images.githubusercontent.com/36507972/48737433-48560000-ec4e-11e8-95b6-012053eefd0d.png">



#### Questions:
N/A
